include/eve/detail/function/simd/riscv/subscript.hpp

```diff
@@ -0,0 +1,112 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/riscv/rvv_common_masks.hpp>
+
+namespace eve::detail
+{
+template<callable_options O, typename T, typename N>
+EVE_FORCEINLINE T
+extract_(EVE_REQUIRES(rvv_), O const&, wide<T, N> v, std::size_t i) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  auto on_first_needed = __riscv_vslidedown_tu(v, v, i, N::value);
+  if constexpr( std::is_floating_point_v<T> ) return __riscv_vfmv_f(on_first_needed);
+  else return __riscv_vmv_x(on_first_needed);
+}
+
+template<callable_options O, typename T, typename N>
+EVE_FORCEINLINE void
+insert_(EVE_REQUIRES(rvv_),
+        O const&,
+        wide<T, N>                & v,
+        std::size_t                 i,
+        std::convertible_to<T> auto x) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  // get mask with 1 on i'th element.
+  logical<wide<T, N>> mask {[i](int j, int) { return j == i; }};
+  v = if_else(mask, static_cast<T>(x), v);
+}
+
+template<callable_options O, typename T, typename N>
+EVE_FORCEINLINE logical<T>
+extract_(EVE_REQUIRES(rvv_), O const&, logical<wide<T, N>> v, std::size_t i) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  constexpr size_t                      array_size = (N::value + 7) / 8;
```

Comment:
N is power of 2. always. you don't need these tricks.

Reply:
This is needed not to be save with non-power of two. I keep this to support cases when n equals 1, 2, 4. In that case I expect the size will be 1. Changed to std::max.